operator/pkg/tasks/init/test_helpers.go

```diff
@@ -0,0 +1,210 @@
+}
+
+// ControlplaneConfig returns the control plane configuration.
+func (t *TestInitData) ControlplaneConfig() *rest.Config {
+	return t.ControlplaneConfigREST
+}
+
+// ControlplaneAddress returns the address of the control plane.
+func (t *TestInitData) ControlplaneAddress() string {
+	return t.ControlplaneAddr
+}
+
+// RemoteClient returns the Kubernetes client for remote interactions.
+func (t *TestInitData) RemoteClient() clientset.Interface {
+	return t.RemoteClientConnector
+}
+
+// KarmadaClient returns the Kubernetes client for interacting with Karmada.
+func (t *TestInitData) KarmadaClient() clientset.Interface {
+	return t.KarmadaClientConnector
+}
+
+// DataDir returns the data directory used by Karmada.
+func (t *TestInitData) DataDir() string {
+	return t.DataDirectory
+}
+
+// CrdTarball returns the CRD tarball used for Karmada installation.
+func (t *TestInitData) CrdTarball() operatorv1alpha1.CRDTarball {
+	return t.CrdTarballArchive
+}
+
+// KarmadaVersion returns the version of Karmada being used.
+func (t *TestInitData) KarmadaVersion() string {
+	return t.KarmadaVersionRelease
+}
+
+// Components returns the Karmada components used in the current installation.
+func (t *TestInitData) Components() *operatorv1alpha1.KarmadaComponents {
+	return t.ComponentsUnits
+}
+
+// FeatureGates returns the feature gates enabled for the current installation.
+func (t *TestInitData) FeatureGates() map[string]bool {
+	return t.FeatureGatesOptions
+}
+
+// AddCert adds a Karmada certificate to the TestInitData.
+func (t *TestInitData) AddCert(cert *certs.KarmadaCert) {
+	t.Certs = append(t.Certs, cert)
+}
+
+// GetCert retrieves a Karmada certificate by its name.
+func (t *TestInitData) GetCert(name string) *certs.KarmadaCert {
+	for _, cert := range t.Certs {
+		parts := strings.Split(cert.CertName(), ".")
+		if parts[0] == name {
+			return cert
+		}
+	}
+	return nil
+}
+
+// CertList returns a list of all Karmada certificates stored in TestInitData.
+func (t *TestInitData) CertList() []*certs.KarmadaCert {
+	return t.Certs
+}
+
+// LoadCertFromSecret loads a Karmada certificate from a Kubernetes secret.
+func (t *TestInitData) LoadCertFromSecret(secret *corev1.Secret) error {
+	if len(secret.Data) == 0 {
+		return fmt.Errorf("cert data is empty")
+	}
+
+	// Dummy implementation: load empty certificate.
+	cert := &certs.KarmadaCert{}
+	t.AddCert(cert)
+	return nil
+}
+
+// createPods creates a specified number of pods in the given namespace
+// with the provided component name.
+//
+// Parameters:
+// - client: Kubernetes client interface.
+// - namespace: Namespace for pod creation.
+// - componentName: Base name for pods and containers.
+// - replicaCount: Number of pods to create.
+// - labels: Labels for pods creation.
+//
+// Returns an error if pod creation fails.
+func createPods(client clientset.Interface, namespace string, componentName string, replicaCount int32, labels map[string]string) error {
```

Comment:
func `createPods` is unused (unused)

Reply:
@XiShanYongYe-Chang It is utilized in the PRs that depend on this PR.

Reply:
More specifically, those are the PRs that utilize `createPods` later: #5632 #5634

Reply:
I seem to have found a loop between these dependencies, so can you help point out which pr to look at first?

Reply:
> Could we ignore this linting issue in this PR?

The lint error of the current pr cannot be ignored. Therefore, if the CI fails, the pr cannot be integrated.

> More specifically, those are the PRs that utilize createPods later: https://github.com/karmada-io/karmada/pull/5632 https://github.com/karmada-io/karmada/pull/5634

Perhaps you can add this method later when you use the createPods method.

Reply:
> Perhaps you can add this method later when you use the createPods method.

Sounds good. I will write the createPods twice in those PRs and we could handle them later by rebasing. 👍

Reply:
@XiShanYongYe-Chang Or What do you think adding this lint directive `// nolint:unused` above createPods and this could be removed later in the dependencies instead of repeating `createPods` twice function ?

Reply:
I tend to wait for the function to be used before filling in, as to whether to write the same function twice, can be refactored to reduce code redundancy.

Reply:
Let's modify this pr first so that subsequent prs can not be blocked.